### PR TITLE
[Contract] Add routers_internal to NetworkConfiguration Entity

### PR DESCRIPTION
**Pull request by haboy52581**
This PR is mainly about:
- Add routers_internal to NetworkConfiguration entity

### CODE REVIEW

File: docs/modules/ROOT/pages/infra_services/data_plane_manager.adoc [hunk 1]
```diff
@@ -178,157 +178,195 @@ anchor:Create_Port_NetworkConfiguration[]
...
+               "hostIp":"10.213.43.187",
+               "portIds":"f37810eb-7f83-45fa-a4d4-1b31e75399d3,f37810eb-7f83-45fa-a4d4-1b31e75399d0"
+            },
+            {
+               "hostIp":"10.213.43.188",
+               "portIds":"f37810eb-7f83-45fa-a4d4-1b31e75399d6,f37810eb-7f83-45fa-a4d4-1b31e75399d7"
+            }
+         ],
+         "dhcpIds":"aaa,bbb",
+         "gatewayMacs":"g1,g2"
```
Comment (xieus) [inline docs/modules/ROOT/pages/infra_services/data_plane_manager.adoc:224]: We only have one GW MAC address.

File: docs/modules/ROOT/pages/infra_services/data_plane_manager.adoc [hunk 2]
```diff
@@ -178,157 +178,195 @@ anchor:Create_Port_NetworkConfiguration[]
...
+   "vpcs_internal":[
+      {
+         "project_id":"3dda2801-d675-4688-a63f-dcda8d327f50",
+         "id":"9192a4d4-ffff-4ece-b3f0-8d36e3d88038",
+         "name":"test_vpc",
+         "description":"",
+         "cidr":"192.168.0.0/16"
+      }
+   ],
+   "subnets_internal":[
```
Comment (xieus) [inline docs/modules/ROOT/pages/infra_services/data_plane_manager.adoc:198]: This subnet_internal is aligned with our InternalSubnetEntity, at least the following fields:
- portIdsPerHost
- gatewayMacs
- dhcpIds

https://github.com/futurewei-cloud/alcor/blob/83f950b8cb1d699761439c75e829da3bb66991b5/web/src/main/java/com/futurewei/alcor/web/entity/dataplane/InternalSubnetEntity.java#L20

File: docs/modules/ROOT/pages/infra_services/data_plane_manager.adoc [hunk 3]
```diff
@@ -178,157 +178,195 @@ anchor:Create_Port_NetworkConfiguration[]
...
+            {
+               "hostIp":"10.213.43.188",
+               "portIds":"f37810eb-7f83-45fa-a4d4-1b31e75399d6,f37810eb-7f83-45fa-a4d4-1b31e75399d7"
+            }
+         ],
+         "dhcpIds":"aaa,bbb",
+         "gatewayMacs":"g1,g2"
+      }
+   ],
+   "routers_internal":[
```
Comment (xieus) [inline docs/modules/ROOT/pages/infra_services/data_plane_manager.adoc:227]: Please update routers_internal with the latest contract in NetworkConfiguration.java.

File: web/src/main/java/com/futurewei/alcor/web/entity/dataplane/NetworkConfiguration.java [hunk 4]
```diff
@@ -46,6 +47,9 @@
   @JsonProperty("neighbor_table")
   private List<NeighborEntry> neighborTable;
 
+  @JsonProperty("routerinfo")
+  private List<InternalRouterInfo> routerTable;
```
Comment (xieus) [inline web/src/main/java/com/futurewei/alcor/web/entity/dataplane/NetworkConfiguration.java:51]: Suggest to rename the variable name to routerInfo.

File: web/src/main/java/com/futurewei/alcor/web/entity/dataplane/NetworkConfiguration.java [hunk 5]
```diff
@@ -46,6 +47,9 @@
...
+
   public void addPortEntity(InternalPortEntity portEntity) {
     if (this.portEntities == null) {
       this.portEntities = new ArrayList<>();
```
Comment (xieus) [inline web/src/main/java/com/futurewei/alcor/web/entity/dataplane/NetworkConfiguration.java:55]: Need to add setter and getter for the new field, and update ToString method.

Comment (haboy52581) [reply web/src/main/java/com/futurewei/alcor/web/entity/dataplane/NetworkConfiguration.java:68]: fixed in a90538a

File: web/src/main/java/com/futurewei/alcor/web/entity/dataplane/NetworkConfiguration.java (hunk 4 above)
Comment (haboy52581) [reply web/src/main/java/com/futurewei/alcor/web/entity/dataplane/NetworkConfiguration.java:51]: fixed in a90538a

File: docs/modules/ROOT/pages/infra_services/data_plane_manager.adoc [hunk 6]
```diff
@@ -552,7 +576,7 @@ Body:
 		"fixed_ips": [{
 			"ip_address": "192.168.2.5",
 			"subnet_id": "a87e0f87-a2d9-44ef-9194-9a62f178594f"
-		}	
+		}
```
Comment (xieus) [inline docs/modules/ROOT/pages/infra_services/data_plane_manager.adoc:579]: Replace "tab" with space throughout the document.

File: web/src/main/java/com/futurewei/alcor/web/entity/dataplane/NetworkConfiguration.java (hunk 4 above)
Comment (xieus) [reply web/src/main/java/com/futurewei/alcor/web/entity/dataplane/NetworkConfiguration.java:51]: @haboy52581 the variable name itself remains unchanged.

**Review by xieus — CHANGES_REQUESTED** [review]

**Comment by xieus** [comment]
@haboy52581 PR description please.

**Review by xieus — APPROVED** [review]

**Pull request merged**